src/test/spec/unified_runner/entity.rs

```diff
@@ -52,6 +53,12 @@ pub(crate) enum Entity {
     None,
 }
 
+#[derive(Debug)]
+pub(crate) struct SessionPtr(pub(crate) *mut ClientSession);
```

Comment:
This is really ugly and I tried a lot of things to avoid it but none of them ended up being feasible.  For the legacy runner, this was avoidable because there's a fixed number of sessions created for the tests and those could just be put into their own structure+lifetime and pulled out of the rest of the test runner.  For the unified runner, sessions are created dynamically, with no statically known number or lifetime.

Storing a raw pointer lets us elide the lifetime; when it's retrieved it's cast to a `&mut ClientSession` with a strictly shorter lifetime than the "real" one, but not in a way that I could prove to the compiler, so ... `unsafe` :(